keps/sig-node/3759-sidecar-containers/README.md

```diff
@@ -0,0 +1,1235 @@
+may be slightly confusing.
+
+  
+#### Alternatives
+
+- Introduce a new collection `infrastructureContainers` that replaces
+  semantically `initContainers`. This collection will allow both - Init
+  containers and sidecar containers. Decision on this alternative can be
+  postponed to after alpha implementation.
+- Introduce a new enum value `UntilPodTermination` for `restartPolicy` on
+  containers. This alternative was rejected because the “Always” semantic is
+  quite easy to explain and potential confusion seems to be minimal.
+- Introduce a new field: “ambient: true” on all containers. This property will
+  make containers be restarted all the time, and will not block the pod
+  completion. This alternative will only work if this field will work similarly
+  for both - Init and regular containers. The only difference will be that for
+  Init containers, readiness of the container will be required before starting
+  the next Init container.
+  - Pros: detaching sidecar proposal from the per-container restartPolicy
+    proposal
+  - Cons: may introduce confusion whether to inject it in the Init list or
+    regular list.
+
+
+### User Stories (Optional)
+
+User stories are covered in Motivation section.
+
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+The following is the list of hypothetical scenarios how users may decide to
+abuse the feature or use it in a non-designed way. This is an exercise to
+understand potential side effects from implementing the feature. We are looking
+for side effects like abuse of the functionality to implement something
+error-prone that sidecars were not designed for. Or causing issues like noisy
+neighbors, etc.
+
+### Scenario 1. User decides to use sidecars as a way to run regular containers
+
+At least one regular container is required by the Pod. Users may decide to run
+all other containers as sidecars and have a single container with the Sleep
+logic inside with timeout or waiting for some cancellation signal from custom
+job orchestration portal. This way users may implement execution with a time
+limit defined by that logic. This is something that users cannot express today
+“declaratively” and will be possible using new sidecar containers. One may
+imagine a scenario when a third party jobs orchestration tool converts all
+containers to sidecar containers programmatically and injects a single job
+orchestration container into the Pod. This can be seen as an abuse of the
+concept, and may lead to issues when pods terminate unexpectedly or Jobs meant
+to be run as regular containers start being run as sidecars being restarted on
+Completion. However we do not see a major problem with it
```

Comment:
I would suggest completing this sentence with a reason why you don't see a problem - "users can already abuse init containers, and it's not an issue" or whatever the best reason was.

Reply:
finished the sentence